### DISCUSSION TITLE
关于 CollectionData 和 TokenDate 的理解这样正确吗

### DISCUSSION THREAD

**Discussion by hi-chenerge** [NonceGeek/Move-Camp#257] [Q&A]
问题1： CollectionData  就是 无聊猿猴，杰伦熊 这种大类

TokenDate 就是无聊猿猴下面的每一个具体编号的NFT

可以这么理解吧

因为我看 CollectionData 里面是一些总的信息,  TokenData 是具体 NFT的信息

问题2：CollectionData 的  supply  有啥用?   是所有 collection 下的 TokenData 的 supply 总和不能超过他吗

```
    struct Collections has key {
        collection_data: Table<String, CollectionData>,
        token_data: Table<TokenDataId, TokenData>,
        create_collection_events: EventHandle<CreateCollectionEvent>,
        create_token_data_events: EventHandle<CreateTokenDataEvent>,
        mint_token_events: EventHandle<MintTokenEvent>,
    }
```

**Comment by qiwihui** [comment]
1. 是的；
2. aptos 的 NFT 协议用的标准是类似于 ERC1155 的标准，就是每一个编号的 NFT 有多个。CollectionData 有两个数量相关的属性，supply 指的是创建的不同种类的 TokenData 的数量，简单理解为已经创建的编号，maximum 是指 TokenData 的最大数量，就是最大的编号：
- 如果 maximum 不为0，则 TokenData 数量 supply 不能大于 maximum
- 如果 maximum 为 0，表示数量不限制，且不追踪数量，即在创建 tokendata 的时候，不会更新supply。

同理，每一个 TokeData 也有 supply 和 maximum 的概念，与上面类似。

并不存在一个属性表示所有 TokenData 下supply 的和或者最大值。

具体可以看源代码： https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token/sources/token.move 中 CollectionData，TokenData 结构的注释，以及 create_tokendata，mint_token 时supply 的变化部分。
